`pytest` fails in a fresh environment with just `pip install ".[test]"` dependencies

I noticed that the project use `pytest`, so I assumed that running `pip install ".[test]"` would install all the dependencies needed to run the full test suite. However, it seems that's not the case _(it's currently complaining that I don't have `vllm`)_. Should I open an issue _(I can add the missing dependency under the `test` section of the `pyproject.toml`)_?

As an afterthought, I checked what's advised in the documentation and implemented in CI. The nice part is that it's straightforward, and I got it right spontaneously:

- https://dottxt-ai.github.io/outlines/latest/community/contribute/
- https://github.com/dottxt-ai/outlines/blob/6a8612bd10525e95a62d9f9c727d3e54d0723f1a/.github/workflows/tests.yml#L32-L49

However, what's less nice is that it fails _(in a clean environment, with Python 3.11)_:

```
pytest
=============================== test session starts ===============================
platform darwin -- Python 3.12.2, pytest-8.3.4, pluggy-1.5.0
benchmark: 5.1.0 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /Users/yvan/outlines
configfile: pyproject.toml
testpaths: tests
plugins: cov-6.0.0, benchmark-5.1.0, mock-3.14.0, anyio-4.2.0
collected 669 items / 1 error                                                     

===================================== ERRORS ======================================
____________ ERROR collecting tests/generate/test_integration_vllm.py _____________
ImportError while importing test module '/Users/yvan/outlines/tests/generate/test_integration_vllm.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/anaconda3/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/generate/test_integration_vllm.py:7: in <module>
    from vllm.sampling_params import SamplingParams
E   ModuleNotFoundError: No module named 'vllm'
============================= short test summary info =============================
ERROR tests/generate/test_integration_vllm.py
!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!
================================ 1 error in 34.65s ================================
pytest  6.38s user 1.33s system 21% cpu 36.662 total
```

On a somewhat related note, what about providing basic **Nix**/**Guix** reproducible developer shell or VSCode **DevContainer**/GitHub **CodeSpace** configuration files for the project? As a side effect, it could make it easier for potential contributors to set up a developer environment!

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -67,7 +67,7 @@ test = [
     "huggingface_hub",
     "openai>=1.0.0",
     "datasets",
-    "vllm; sys_platform != 'darwin'",
+    "vllm; sys_platform == 'linux'",
     "transformers",
     "pillow",
     "exllamav2",

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,0 +1,14 @@
+import sys
+
+import pytest
+
+
+def pytest_collection_modifyitems(config, items):
+    if sys.platform != "linux":
+        skip_vllm = pytest.mark.skip(reason="vLLM models can only be run on Linux.")
+        for item in items:
+            if "test_integration_vllm" in item.nodeid:
+                item.add_marker(skip_vllm)
+                print(
+                    f"WARNING: {item.nodeid} is skipped because vLLM only supports Linux platform (including WSL)."
+                )

--- a/tests/generate/test_integration_vllm.py
+++ b/tests/generate/test_integration_vllm.py
@@ -4,7 +4,11 @@ import re
 import pytest
 import torch
 from pydantic import BaseModel, constr
-from vllm.sampling_params import SamplingParams
+
+try:
+    from vllm.sampling_params import SamplingParams
+except ImportError:
+    pass
 
 import outlines.generate as generate
 import outlines.grammars as grammars

